File: Source/ArticlesService/Models/Article.cs
using System;
using System.Collections.Generic;
// ReSharper disable ClassWithVirtualMembersNeverInherited.Global

namespace ArticlesService.Models
{
    /// <summary>
    /// Модель статьи.
    /// </summary>
    public class Article
    {
        /// <summary>
        /// Идентификатор.
        /// </summary>
        public virtual long Id { get; set; }

        /// <summary>
        /// Заголовок.
        /// </summary>
        public virtual string Title { get; set; }

        /// <summary>
        /// Автор.
        /// </summary>
        public virtual string Author { get; set; }

        /// <summary>
        /// Текст.
        /// </summary>
        public virtual string Content { get; set; }

        /// <summary>
        /// Дата создания.
        /// </summary>
        public virtual DateTime Created { get; set; }

        /// <summary>
        /// Комментарии к статье
        /// </summary>
        public virtual ICollection<Comment> Comments { get; set; }
    }
}

File: Source/ArticlesWeb/ArticlesApp/src/app/models/article.ts
import { ArticleComment } from "./article.comment";

/**
 * Модель статьи.
 */
export class Article {
  /**
   * Идентификатор.
   */
  id: number = 0;

  /**
   * Заголовок.
   */
  title: string;

  /**
   * Автор.
   */
  author: string;

  /**
   * Текст.
   */
  content: string;

  /**
   * Дата создания.
   */
  created: Date = new Date();

  /**
   * Комментарии к статье
   */
  comments: ArticleComment[] = [];
}
